core/src/vcs/git.ts

```
@@ -417,4 +409,30 @@ export class GitHandler extends VcsHandler {
     }
     return undefined
   }
+
+  async getBranchName(log: LogEntry, path: string): Promise<string | undefined> {
+    const git = this.gitCli(log, path)
+    try {
+      return (await git("rev-parse", "--abbrev-ref", "HEAD"))[0]
+    } catch (err) {
+      if (err.exitCode === 128) {
+        try {
+          // If this doesn't throw, then we're in a repo with no commits.
+          await git("rev-list", "-n", "1", "--all")
+          return undefined
+        } catch (notInRepoError) {
+          // Throw nice error when we detect that we're not in a repo root
+          throw new RuntimeError(notInRepoRootErrorMessage(path), { path })
```

(Oops, meant to edit and not delete what I previously wrote)

It seems we also need a more specific error check for this nested git command. Or perhaps we should generally fail silently here, and just document that `${git.branch}` may be undefined in certain cases? We might for example be on a detached HEAD (e.g. rebasing).